18-websockets/classworks/Gavruk_Alexander/src/pages/Chat/Chat.jsx

```diff
@@ -0,0 +1,129 @@
+import React, { useEffect, useState } from "react";
+import socketIOClient from "socket.io-client";
+
+import './Chat.css';
+import Message from './components/Message';
+
+const ENDPOINT = "http://localhost:3001/";
+const socket = socketIOClient(ENDPOINT);
+
+function Chat() {
+  const [usernameInputValue, setUsernameInputValue] = useState('');
+  const [msgInputValue, setMsgInputValue] = useState('');
+  const [username, setUsername] = useState('');
+  const [messages, setMessages] = useState([]);
+  const [typingUsers, setTypingUsers] = useState([]);
+  const [timerId, setTimerId] = useState(null);
+
+  const onUsernameSubmit = (username) => {
+    socket.emit('set_username', username);
+    setUsernameInputValue('');
+  }
+
+  const onSendMessage = (msg) => {
+    socket.emit('new_message', { username, value: msg });
+    setMsgInputValue('');
+  }
+
+  const onType = (username) => {
+    const timerId = setTimeout(() => {
+      setTypingUsers((typingUsers) => typingUsers.filter((name) => name !== username))
+    }, 7000);
+    setTimerId(timerId);
```

Comment:
looks like you rewrite timerId and old typing users won't disappear. You need to save timerId for every user and clear them if you receive `typing` event of the user again.